File: Code/26.remove-duplicates-from-sorted-array.cpp
/*
 * @lc app=leetcode id=26 lang=cpp
 *
 * [26] Remove Duplicates from Sorted Array
 */

// @lc code=start
#include <vector>
using namespace std;

class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        if (nums.size() == 0)
        {
            return 0;
        }
        else {
            int slow = 1;
            for (int fast = 1; fast < nums.size(); ++fast )
            {
                if(nums[fast] != nums[slow-1]){
                    nums[slow] = nums[fast];
                    ++slow;
                };
            }
            return slow;
        }
    }
};
// @lc code=end

File: 26.remove-duplicates-from-sorted-array.py
#
# @lc app=leetcode id=26 lang=python3
#
# [26] Remove Duplicates from Sorted Array
#

# @lc code=start
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        if len(nums)==1:
            return 1
        slow = 0
        quick = 0
        while quick!=len(nums):
            if nums[slow] != nums[quick]:
                slow+=1
                nums[slow]= nums[quick]
            quick+=1
        return slow+1

        
# @lc code=end

